components/filters.tsx

```diff
@@ -57,39 +59,77 @@ export const Filters = () => {
     );
   };
 
+  function loadFilters(data: { [x: string]: any; }) {
+    // set defaults or load from arg
+    const setData = {
+      min_sa: data["min_sa"] ?? 0,
+      max_sa: data["max_sa"] ?? 99,
+      min_drivers: data["min_cd"] ?? 0,
+      max_drivers: data["max_cd"] ?? 99,
+      min_tm: data["min_tm"] ?? 0,
+      max_tm: data["max_tm"] ?? 3,
+      showEmpty: data["showEmpty"] ?? false ? true : false,
+      showFull: data["showFull"] ?? true ? true : false,
+      class: data["class"] ?? ["mixed", "gt3", "gt4", "gtc", "tcx", "gt2"],
+      dlc: data["dlc"] ?? [
+        "base",
+        "icgt",
+        "gtwc",
+        "bgt",
+        "atp",
+        "gt2",
+        "nurb-24",
+      ],
+      track: data["track"] ?? Object.keys(staticData.tracks),
+      sessions: data["session"] ?? ["race", "qualifying", "practice"],
+      favouritesOnly: data["favouritesOnly"] ?? false ? true : false,
+    };
+    setFilterData(setData);
+  }
+
   useEffect(() => {
     if (filterData) {
       localStorage.setItem("filters", JSON.stringify(filterData));
       ctx.refetch(filterData);
     } else {
-      // set defaults or load from localStorage
       const data = JSON.parse(localStorage.getItem("filters") ?? "{}");
-      const setData = {
-        min_sa: data["min_sa"] ?? 0,
-        max_sa: data["max_sa"] ?? 99,
-        min_drivers: data["min_cd"] ?? 0,
-        max_drivers: data["max_cd"] ?? 99,
-        min_tm: data["min_tm"] ?? 0,
-        max_tm: data["max_tm"] ?? 3,
-        showEmpty: data["showEmpty"] ?? false ? true : false,
-        showFull: data["showFull"] ?? true ? true : false,
-        class: data["class"] ?? ["mixed", "gt3", "gt4", "gtc", "tcx", "gt2"],
-        dlc: data["dlc"] ?? [
-          "base",
-          "icgt",
-          "gtwc",
-          "bgt",
-          "atp",
-          "gt2",
-          "nurb-24",
-        ],
-        sessions: data["session"] ?? ["race", "qualifying", "practice"],
-        favouritesOnly: data["favouritesOnly"] ?? false ? true : false,
-      };
-      setFilterData(setData);
+      loadFilters(data);
     }
   }, [filterData]);
 
+  const selectAllTracks = () => {
+    let tracks = Object.keys(staticData.tracks);
+    localStorage.setItem("filters", JSON.stringify(tracks))
+    loadFilters({ track: tracks})
+  }
+
+  const selectNoneTracks = () => {
+    localStorage.setItem("filters", "{}")
+    loadFilters({ track: [] })
+  }
```

Comment:
I think this needs a rework, setting the localstorage will overwrite all other filters as well. What you'd probably want to do is something like (but not exactly): 
```js
if(none) {
// Empty for no tracks
filterData.track = [];
} else if(all) {
// All tracks
filterData.track = Object.keys(staticData.tracks);
}
setFilterData(filterData);
```

Then the `useEffect` will handle updating the localstorage and updating the servers. (So `loadFilters` should not be called from anywhere else, I don't mind still separating it out as a function though.)